src/vm/customattribute.cpp

```diff
@@ -166,8 +166,15 @@ CustomAttributeManagedValues Attribute::GetManagedCaValue(CaValue* pCaVal)
         BOOL bAllBlittableCa = arrayType != SERIALIZATION_TYPE_ENUM;
 
         if (length == (ULONG)-1)
+        {
+            GCPROTECT_BEGIN(gc)
+            {   
+                if (arrayType == SERIALIZATION_TYPE_ENUM)
```

Comment:
Sorry, I'm a bit lost here... there is a first `return gc;` after `if (length == (ULONG)-1)` ... can it be included in the merged `GCPROTECT_BEGIN { ... } GCPROTECT_END();` or should it disappear by changing the if statement etc..? 

and also (sorry for another trivial question) ... 
when "if (arrayType == SERIALIZATION_TYPEENUM)" statement is hoisted up before the "if (length == (ULONG)-1)", do you agree that executing the "for (COUNT_T i = 0; i < length; i ++)
            Attribute::SetBlittableCaValue" *after* it instead of *before* is irrelevant?